src/components/Services/ServiceDetails.jsx

```diff
@@ -1,103 +1,138 @@
 import { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
 import { reviews as fetchReviews } from "../Reviews/reviews.jsx";
-import { services } from "./services.jsx";
-import { slugify } from "./utils.jsx";
-import {ReviewSection} from "../Reviews/ReviewSection.jsx"
+import { BACKEND_URL } from "../../config/api.js";
+import { ReviewSection } from "../Reviews/ReviewSection.jsx";
+import axios from "axios";
+import "@fortawesome/fontawesome-free/css/all.min.css";
 function ServiceDetails() {
-  const { slug } = useParams();
-
+  const { id } = useParams();
   const [service, setService] = useState(null);
   const [loading, setLoading] = useState(true);
   const [error] = useState(null);
   const [reviews, setReviews] = useState([]);
 
   useEffect(() => {
-    const foundService = services.find((s) => slugify(s.title) === slug);
+    fetchServiceDetails();
+  }, [id]);
 
-    setService(foundService || null);
-    console.log("service", foundService);
-    setLoading(false);
-    if (foundService) {
-      const serviceWithReviews = fetchReviews.filter(
-        (review) => review.serviceId === foundService.id
-      );
-      setReviews(serviceWithReviews);
-      console.log("reviews", serviceWithReviews);
+  const fetchServiceDetails = async () => {
+    try {
+      setLoading(true);
+      const res = await axios.get(`${BACKEND_URL}/service/${id}`);
+      const serviceData = Array.isArray(res.data) ? res.data[0] : res.data;
+      setService(serviceData);
+      setReviews(res.data[0].Reviews);
+      console.log("service data", res.data);
+    } catch (error) {
+      console.error("failed fetching data of this service", error);
```

Comment:
Thought: It's good that we are catching the error here and displaying the error coming from the backend. I am wondering is it better to display the raw error coming from the backend or to 

Suggestion: We should implement an alert component that can be displayed to the user. Let's implement it in a separate PR

Reply:
I have created the issue here: https://github.com/Germany-Assist/germany-assist-react/issues/26